src/libsync/connectionvalidator.cpp

```diff
@@ -252,7 +253,17 @@ void ConnectionValidator::slotUserFetched(const QVariantMap &json)
     QString user = json.value("ocs").toMap().value("data").toMap().value("id").toString();
     if (!user.isEmpty()) {
         _account->setDavUser(user);
+
+        AvatarJob *job = new AvatarJob(_account, this);
+        QObject::connect(job, SIGNAL(avatarPixmap(QPixmap)), this, SLOT(slotAvatarPixmap(QPixmap)));
+
```

Comment:
I guess capabilities fetching could run in parallel to fetching the username, and following on that, the avatar, because the avatar needs the username. Maybe the fetch job of the avatar (at least) should be owned by the account, and could become fully async?